judgels-frontends/raphael/src/routes/uriel/contests/single/supervisors/ContestSupervisorAddForm/ContestSupervisorAddForm.tsx

```diff
@@ -1,33 +1,109 @@
-import { Button, Intent } from '@blueprintjs/core';
+import { Button, Intent, HTMLTable } from '@blueprintjs/core';
 import * as React from 'react';
 import { Field, InjectedFormProps, reduxForm } from 'redux-form';
 
 import { Required } from 'components/forms/validations';
 import { FormTextArea } from 'components/forms/FormTextArea/FormTextArea';
+import { FormTableInput } from 'components/forms/FormTableInput/FormTableInput';
+import { FormCheckbox } from 'components/forms/FormCheckbox/FormCheckbox';
+import { SupervisorManagementPermission } from 'modules/api/uriel/contestSupervisor';
 
 export interface ContestSupervisorAddFormData {
   usernames: string;
+  grantAllPermissions?: boolean;
+  grantedPermissions?: { [key: string]: boolean };
 }
 
 export interface ContestSupervisorAddFormProps extends InjectedFormProps<ContestSupervisorAddFormData> {
   renderFormComponents: (fields: JSX.Element, submitButton: JSX.Element) => JSX.Element;
 }
 
-const ContestSupervisorAddForm = (props: ContestSupervisorAddFormProps) => {
-  const usernamesField: any = {
-    name: 'usernames',
-    label: 'Usernames',
-    labelHelper: '(one username per line, max 100 users)',
-    rows: 20,
-    validate: [Required],
-    autoFocus: true,
+interface ContestSupervisorAddFormState {
+  grantAllPermissions: boolean;
```

Comment:
Let's simplify and not use "all" checkbox in this form. It will hide the individual permissions to the manager, which is crucial.

(It's kinda different from the "all" checkbox in grading languages option in contest settings, because that just means we don't care about the programming language.)